pkg/kubelet/kubelet.go

```diff
@@ -2703,7 +2703,7 @@ func (kl *Kubelet) handlePodResourcesResize(pod *v1.Pod) {
 			//TODO(vinaykul,InPlacePodVerticalScaling): Can we recover from this in some way? Investigate
 			klog.ErrorS(err, "SetPodAllocation failed", "pod", klog.KObj(pod))
 		}
-		*pod = *updatedPod
+		pod = updatedPod
```

Comment:
split this (and the unit test) into a separate commit, since it's different than the volume manager race?

Reply:
Yes. Indeed, this PR fixed 3 data race issues.

Split.